### ISSUE TITLE
test failure after adding the nvptx test cases

### ISSUE THREAD

**Issue by andrewrk** [ziglang/zig#10968]
### Zig Version

0.10.0-dev.889+6dc5ce931

### Steps to Reproduce

Run the stage2 tests on master branch:

```
[nix-shell:~/Downloads/zig/build-release]$ ./zig build test-stage2 -fqemu -Denable-llvm -fwasmtime -Dskip-compile-errors
```

This diff works around the issue:

```diff
--- a/test/cases.zig
+++ b/test/cases.zig
@@ -16,5 +16,5 @@ pub fn addCases(ctx: *TestContext) !void {
     try @import("stage2/riscv64.zig").addCases(ctx);
     try @import("stage2/plan9.zig").addCases(ctx);
     try @import("stage2/x86_64.zig").addCases(ctx);
-    try @import("stage2/nvptx.zig").addCases(ctx);
+    //try @import("stage2/nvptx.zig").addCases(ctx);
 }
```

cc @gwenzek 

### Expected Behavior

All tests passed.

### Actual Behavior

```
Test [1/105] test ""... compiler [195/197] simple addition and subtraction [1/1] update [2/3] Segmentation fault at address 0x0
/home/andy/Downloads/zig/src/codegen/llvm.zig:277:28: 0x2e71e11 in codegen.llvm.Object.deinit (test)
        self.target_machine.dispose();
                           ^
/home/andy/Downloads/zig/src/codegen/llvm.zig:287:20: 0x2cecd98 in codegen.llvm.Object.destroy (test)
        self.deinit(gpa);
                   ^
/home/andy/Downloads/zig/src/link/NvPtx.zig:69:29: 0x2ae2b76 in link.NvPtx.deinit (test)
    self.llvm_object.destroy(self.base.allocator);
                            ^
/home/andy/Downloads/zig/src/link.zig:584:30: 0x2ad84e8 in link.File.destroy (test)
                parent.deinit();
                             ^
/home/andy/Downloads/zig/src/Compilation.zig:2023:30: 0x2a9a4c4 in Compilation.update (test)
        comp.bin_file.destroy();
                             ^
/home/andy/Downloads/zig/src/Compilation.zig:4914:31: 0x2d2ab6d in Compilation.buildOutputFromZig (test)
    try sub_compilation.update();
                              ^
/home/andy/Downloads/zig/src/Compilation.zig:3065:36: 0x2b28206 in Compilation.processOneJob (test)
            comp.buildOutputFromZig(
                                   ^
/home/andy/Downloads/zig/src/Compilation.zig:2676:30: 0x2b0ff8d in Compilation.performAllTheWork (test)
            try processOneJob(self, work_item, main_progress_node);
                             ^
/home/andy/Downloads/zig/src/Compilation.zig:2082:31: 0x2a9add4 in Compilation.update (test)
    try comp.performAllTheWork();
                              ^
/home/andy/Downloads/zig/src/test.zig:937:28: 0x2a7efe0 in TestContext.runOneCase (test)
            try comp.update();
                           ^
/home/andy/Downloads/zig/src/test.zig:667:23: 0x2a6afd2 in TestContext.run (test)
            runOneCase(
                      ^
/home/andy/Downloads/zig/src/test.zig:35:16: 0x2a46223 in test "" (test)
    try ctx.run();
               ^
/home/andy/Downloads/zig/lib/std/special/test_runner.zig:77:28: 0x2a6ba30 in std.special.main (test)
        } else test_fn.func();
                           ^
/home/andy/Downloads/zig/lib/std/start.zig:551:22: 0x2e5d6ec in std.start.callMain (test)
            root.main();
                     ^
/home/andy/Downloads/zig/lib/std/start.zig:495:12: 0x2a6d9a7 in std.start.callMainWithArgs (test)
    return @call(.{ .modifier = .always_inline }, callMain, .{});
           ^
/home/andy/Downloads/zig/lib/std/start.zig:460:12: 0x2a6d752 in std.start.main (test)
    return @call(.{ .modifier = .always_inline }, callMainWithArgs, .{ @intCast(usize, c_argc), c_argv, envp });
           ^
error: the following test command crashed:
/home/andy/Downloads/zig/zig-cache/o/8f999b43aee63c8663facd19295aa7d5/test /home/andy/Downloads/zig/build-release/zig
test...The following command exited with error code 1:
/home/andy/Downloads/zig/build-release/zig test /home/andy/Downloads/zig/src/test.zig -lc /home/andy/Downloads/zig/build-release/zigcpp/libzigcpp.a /home/andy/local/llvm13-release/lib/libclangFrontendTool.a /home/andy/local/llvm13-release/lib/libclangCodeGen.a /home/andy/local/llvm13-release/lib/libclangFrontend.a /home/andy/local/llvm13-release/lib/libclangDriver.a /home/andy/local/llvm13-release/lib/libclangSerialization.a /home/andy/local/llvm13-release/lib/libclangSema.a /home/andy/local/llvm13-release/lib/libclangStaticAnalyzerFrontend.a /home/andy/local/llvm13-release/lib/libclangStaticAnalyzerCheckers.a /home/andy/local/llvm13-release/lib/libclangStaticAnalyzerCore.a /home/andy/local/llvm13-release/lib/libclangAnalysis.a /home/andy/local/llvm13-release/lib/libclangASTMatchers.a /home/andy/local/llvm13-release/lib/libclangAST.a /home/andy/local/llvm13-release/lib/libclangParse.a /home/andy/local/llvm13-release/lib/libclangSema.a /home/andy/local/llvm13-release/lib/libclangBasic.a /home/andy/local/llvm13-release/lib/libclangEdit.a /home/andy/local/llvm13-release/lib/libclangLex.a /home/andy/local/llvm13-release/lib/libclangARCMigrate.a /home/andy/local/llvm13-release/lib/libclangRewriteFrontend.a /home/andy/local/llvm13-release/lib/libclangRewrite.a /home/andy/local/llvm13-release/lib/libclangCrossTU.a /home/andy/local/llvm13-release/lib/libclangIndex.a /home/andy/local/llvm13-release/lib/libclangToolingCore.a /home/andy/local/llvm13-release/lib/liblldDriver.a /home/andy/local/llvm13-release/lib/liblldMinGW.a /home/andy/local/llvm13-release/lib/liblldELF.a /home/andy/local/llvm13-release/lib/liblldCOFF.a /home/andy/local/llvm13-release/lib/liblldWasm.a /home/andy/local/llvm13-release/lib/liblldMachO.a /home/andy/local/llvm13-release/lib/liblldReaderWriter.a /home/andy/local/llvm13-release/lib/liblldCore.a /home/andy/local/llvm13-release/lib/liblldYAML.a /home/andy/local/llvm13-release/lib/liblldCommon.a -lLLVMWindowsManifest -lLLVMXRay -lLLVMLibDriver -lLLVMDlltoolDriver -lLLVMCoverage -lLLVMLineEditor -lLLVMXCoreDisassembler -lLLVMXCoreCodeGen -lLLVMXCoreDesc -lLLVMXCoreInfo -lLLVMX86Disassembler -lLLVMX86AsmParser -lLLVMX86CodeGen -lLLVMX86Desc -lLLVMX86Info -lLLVMWebAssemblyDisassembler -lLLVMWebAssemblyAsmParser -lLLVMWebAssemblyCodeGen -lLLVMWebAssemblyDesc -lLLVMWebAssemblyUtils -lLLVMWebAssemblyInfo -lLLVMSystemZDisassembler -lLLVMSystemZAsmParser -lLLVMSystemZCodeGen -lLLVMSystemZDesc -lLLVMSystemZInfo -lLLVMSparcDisassembler -lLLVMSparcAsmParser -lLLVMSparcCodeGen -lLLVMSparcDesc -lLLVMSparcInfo -lLLVMRISCVDisassembler -lLLVMRISCVAsmParser -lLLVMRISCVCodeGen -lLLVMRISCVDesc -lLLVMRISCVInfo -lLLVMPowerPCDisassembler -lLLVMPowerPCAsmParser -lLLVMPowerPCCodeGen -lLLVMPowerPCDesc -lLLVMPowerPCInfo -lLLVMNVPTXCodeGen -lLLVMNVPTXDesc -lLLVMNVPTXInfo -lLLVMMSP430Disassembler -lLLVMMSP430AsmParser -lLLVMMSP430CodeGen -lLLVMMSP430Desc -lLLVMMSP430Info -lLLVMMipsDisassembler -lLLVMMipsAsmParser -lLLVMMipsCodeGen -lLLVMMipsDesc -lLLVMMipsInfo -lLLVMLanaiDisassembler -lLLVMLanaiCodeGen -lLLVMLanaiAsmParser -lLLVMLanaiDesc -lLLVMLanaiInfo -lLLVMHexagonDisassembler -lLLVMHexagonCodeGen -lLLVMHexagonAsmParser -lLLVMHexagonDesc -lLLVMHexagonInfo -lLLVMBPFDisassembler -lLLVMBPFAsmParser -lLLVMBPFCodeGen -lLLVMBPFDesc -lLLVMBPFInfo -lLLVMAVRDisassembler -lLLVMAVRAsmParser -lLLVMAVRCodeGen -lLLVMAVRDesc -lLLVMAVRInfo -lLLVMARMDisassembler -lLLVMARMAsmParser -lLLVMARMCodeGen -lLLVMARMDesc -lLLVMARMUtils -lLLVMARMInfo -lLLVMAMDGPUDisassembler -lLLVMAMDGPUAsmParser -lLLVMAMDGPUCodeGen -lLLVMAMDGPUDesc -lLLVMAMDGPUUtils -lLLVMAMDGPUInfo -lLLVMAArch64Disassembler -lLLVMAArch64AsmParser -lLLVMAArch64CodeGen -lLLVMAArch64Desc -lLLVMAArch64Utils -lLLVMAArch64Info -lLLVMOrcJIT -lLLVMMCJIT -lLLVMJITLink -lLLVMInterpreter -lLLVMExecutionEngine -lLLVMRuntimeDyld -lLLVMOrcTargetProcess -lLLVMOrcShared -lLLVMDWP -lLLVMSymbolize -lLLVMDebugInfoPDB -lLLVMDebugInfoGSYM -lLLVMOption -lLLVMObjectYAML -lLLVMMCA -lLLVMMCDisassembler -lLLVMLTO -lLLVMPasses -lLLVMCFGuard -lLLVMCoroutines -lLLVMObjCARCOpts -lLLVMipo -lLLVMVectorize -lLLVMLinker -lLLVMInstrumentation -lLLVMFrontendOpenMP -lLLVMFrontendOpenACC -lLLVMExtensions -lLLVMDWARFLinker -lLLVMGlobalISel -lLLVMMIRParser -lLLVMAsmPrinter -lLLVMDebugInfoMSF -lLLVMDebugInfoDWARF -lLLVMSelectionDAG -lLLVMCodeGen -lLLVMIRReader -lLLVMAsmParser -lLLVMInterfaceStub -lLLVMFileCheck -lLLVMFuzzMutate -lLLVMTarget -lLLVMScalarOpts -lLLVMInstCombine -lLLVMAggressiveInstCombine -lLLVMTransformUtils -lLLVMBitWriter -lLLVMAnalysis -lLLVMProfileData -lLLVMObject -lLLVMTextAPI -lLLVMMCParser -lLLVMMC -lLLVMDebugInfoCodeView -lLLVMBitReader -lLLVMCore -lLLVMRemarks -lLLVMBitstreamReader -lLLVMBinaryFormat -lLLVMTableGen -lLLVMSupport -lLLVMDemangle -lz /nix/store/4dymr3spya676kdp21r7mwcrj806w0ah-gcc-10.3.0/lib64/libstdc++.a -lunwind --cache-dir /home/andy/Downloads/zig/zig-cache --global-cache-dir /home/andy/.cache/zig --name test --pkg-begin test_cases /home/andy/Downloads/zig/test/cases.zig --pkg-end --pkg-begin build_options /home/andy/Downloads/zig/zig-cache/options/jYihacv9LoDsse9jeZgvMXCDXZx8yMXiq9i6nVzKsvnrOlq6JQ2lnpde3h1J338D --pkg-end -I /home/andy/local/llvm13-release/include -L /home/andy/local/llvm13-release/lib -isystem /home/andy/local/llvm13-release/include --enable-cache 
```

**Comment by gwenzek** [comment]
Thanks for flagging, and sorry for the bug. 

I'm pretty sure I managed to run the tests successfully at some point, and that they also ran in CI. 
but with the current master I'm seeing several issues:

1. a bug in link/Nvptx.zig where I'll sometimes do a double free
2. LLVM-Nvptx backend complain about aliases: `LLVM ERROR: Module has aliases, which NVPTX does not support.`

Aliases seems to be generated by "double exports" coming from:
https://github.com/ziglang/zig/blob/628e9e6d040979bd0a2cba05e854014dee5a7d55/lib/std/special/compiler_rt.zig#L97-L106
 
I'm not sure what the "is_test" mean, is it supposed to be false when running the stage2 tests that compile sample code ?

3. Disabling the double exports I get an error while compiling __extenddftf2 itself:

``` LLVM ERROR: Cannot select: 0xa8e1f98: ch = NVPTXISD::StoreRetval<(store (s128), align 1)> 0xa34cc88, Constant:i32<0>, 0xa8e2340
  0xa8e2410: i32 = Constant<0>
  0xa8e2340: i64,ch = CopyFromReg 0xa34cc88, Register:i64 %8
    0xa8e28f0: i64 = Register %8
In function: __extenddftf2
```

Why do I get the error when running `test-stage2` but not when using stage2 to compile my code for NVPTX platform ? 
Could someone give me more context about what `compiler_rt.zig` does ? 
and how test-stage2 compilation differ from regular stage2 compilation ?

In the meantime I'll try a bisect to better understand what's going on.

**Comment by gwenzek** [comment]
I think part of the issue is that I'm using `exeUsingLlvmBackend` to create my test cases,
which according to this comment will pull up all libc, which I  don't need.

https://github.com/ziglang/zig/blob/628e9e6d040979bd0a2cba05e854014dee5a7d55/src/test.zig#L324-L326

I do need the `options.use_llvm`,  but currently `link_libc` and `use_llvm` are tied together in `test.zig`:

https://github.com/ziglang/zig/blob/628e9e6d040979bd0a2cba05e854014dee5a7d55/src/test.zig#L887-L895

Maybe I would need to split that up, but that's not enough to solve my bug.
Not linking libc yields a link error `ld.lld: error: undefined symbol: __lttf2`, because of course I've disabled the alias `__lttf2` earlier.

So it seems that somehow the generated test needs `__lttf2` which is a float128 comparison function ?
I must be getting something wrong in how I'm creating the test case.

**Comment by gwenzek** [comment]
I'm proposing #11028  to fix the tests.
The main issue was that the test runner was trying to build and link libc for NVPTX platform which wasn't what I intended.
So I've added the option for test cases to disable linking libc, as it was suggested in a comment in `test.zig`

**Comment by andrewrk** [comment]
Another test failure due to the nvptx test cases:

```
[nix-shell:~/Downloads/zig/build-release]$ ./zig build test-stage2 -fqemu -Denable-llvm -fwasmtime -Dskip-compile-errors
Test [1/105] test ""... compiler [1/196] stage2 compile errors [1/2] update [2/3] AST Lowering [371] c/compiler [65/196] simple addition and subtraction [1/1] update [2/3] Compile C Objects [173/1341] pthrecompiler [194/196] nvptx: simple addition and subtraction [1/1] update [2/3] malloc(): invalid size (unsorted)
error: the following test command crashed:
/home/andy/Downloads/zig/zig-cache/o/90d04a6adf65fa328f95b22d5ec6c13e/test /home/andy/Downloads/zig/build-release/zig
```

**Comment by gwenzek** [comment]
This one seems more serious. I do have a stack trace leading into LLVM ptx backend. I need to investigate.

Also I can't compile anymore zig code to ptx. It seems to be related to the new support for the LLVM debut info. The default ptx version doesn't have debug info, I've tried to explicitly set a more recent version through command line, but it doesn't seem to propagate to the backend. This should be simpler to fix.

It's hard to keep up with your insane progress rate ^^'

**Comment by andrewrk** [comment]
> The default ptx version doesn't have debug info

In the frontend we can set `strip` to `true` for the ptx backend, then no debug info will make it into the LLVM module :+1: 



> It's hard to keep up with your insane progress rate ^^'

No need to. Take your time with the ptx backend and we'll get it sorted out eventually :)

**Comment by gwenzek** [comment]
So there are two issues at hand here:

* PTX doesn't support debug info until PTX 7.5. This version will be supported by LLVM 14 onward, I can simply disable debug info for PTX in the meantime like we do for Wasm.

* A bug in LLVM 13 PTX backend that seems to have been fixed in LLVM trunk: [godbolt](https://godbolt.org/#z:OYLghAFBqd5QCxAYwPYBMCmBRdBLAF1QCcAaPECAMzwBtMA7AQwFtMQByARg9KtQYEAysib0QXACx8BBAKoBnTAAUAHpwAMvAFYTStJg1DIApACYAQuYukl9ZATwDKjdAGFUtAK4sGIM6SuADJ4DJgAcj4ARpjEIADMXKQADqgKhE4MHt6%2B/ilpGQIhYZEsMXGJtpj2jgJCBEzEBNk%2Bfkl2mA6Z9Y0ExRHRsXoKDU0tue2jfaEDZUNcAJS2qF7EyOwcAPSbANQAKgh4CjtUXgxdAjvMeABu1QCeO3gsyfRsgscECJg7aC%2BNR0uqCoOwIAHdUCc8KpMOgdukAF4/RrEJj3Y7Aq6DYgKEwaACC21B3x2KLRT2OySYCiUcKijyYO1SoQIsVJDDhjPiZgAtFFCBiqFQlAQAHR4wm7A5HCmk36oFgsIHJRwsPAIpi1BigyEKJh3HZsJXER5g77a%2B4rHYCWiPMKw%2BEsMS0UnEVH3cUEokSokHWLI4g/cGQ56vTDvBpa61hUg7LzpIxys7pYBhOHcvmEa1CkU%2B3aGOECH7x0LAOUptPZ4WYMV5/Yk8xmZN4VOwxuy4Gs7XoSExUTxoMk4HVgiG%2BOjmJMgqOA2Ge5gtF1ixeUdfH7ARixTUO6kseWCJihY6YVQEVE7Rv0JjtgsXsxmJjodDto9nrwXBgKT2SnYASV/chuMSQZiAA1heXpSocxwyugXhBpCa4nEwDjEpqTzam41jWNadzEFQtCoGCpbwi2lYspgG7ELKcg2HWZp0D8qB4QRREkc2rZwrBmA0JWMQIPqTjUV86FgqiyTHI0KwchKLKGuGyAvAA%2BhxaYQCyABUOzoJqTBJBhBBaTpDQBHGn5kQ6cl4PpqmWYITxmAsF4AOxWASOweU8IIQMZekmAArBY1kBQAIjsYBgCY8Rhb5ZgBUFcX%2BSFCwSp5aU7IGBCrN2ulcPFwVJRe8RAbF%2BWJSFUVufiaXYZYwXxBYd5WHVcUNalnl4N5vl5YFBVhRFUUxbpiUJaFKXuelHmZdl2m5flPVhVFJXDWVoWVe1Hm1UFeUNU11h4K1VVpdNxA5Q0PXbaFRVYdFs0matSXrQSJjORVz2QfWMobmEqKsp8JLUjS4ZRLaOxgj8EJeLQcK0HgoEIfCmA/FEK5xkoxwVnZrJURhIyYI%2Bsn2UainJEpmPoOpghGblsaaXdTCmVZ%2BlWY5LlHR1XVzb1C3hZFt2lb15XjdVk0ZTWM3dfNV1LfTI0HWtbUTZtlj7TtjVbfLisix5nU7D5XOXYVA38ytgtjRt6UnWdfnc9LxWyw9FVazVKt1Wre0tU92ti1lp30xdfXXYNDtm49WsvSFHBLLQnD%2BbwfgcFopCoJwWGu41CgrGsPzmIkvAEJoUekBASBoFguCECQ5CUDQ9DMGwnA8DIggiGI7BSM38hKGohekLoAQGEYIC0LQNwsFUNSZC4HLjH4ATBDMpTlBISSpOkWqz3ka%2BFAw/RL/M7TVJ0Wo9GMnitHkHQfqf0wlNiK%2B2FMm8BCMvR7/fXCLMsqzrJwlU7AAWQwFDTAv5Fq3TABwP6BAlKiCUJAiUmcf6YCUrXRgrAc63UbNA2B1JMCNglKMDco5fIGEtKjYOjZMA8jwAANkkCAehNCuBmAABwSDYTyG4XBaESFoVw7kCReQMB4UIxhkgCEEiITWUExA8BhiKote8DAbgqlUEws4oEGBEQYDyZAXgdI8nxvySR%2BIJQAAEmAaEUQZWIzAXRnHrrCJSj50DUWAIRKIYg2ZPFobGF66tApcGcjsVQTxWHS1ei5RarlfE7A0P42J8VgmhPCZEsKZwsAggjrGMQZEdgSIJJYpSeBWE2IoqdbxjiMHoBcU%2BdxnjvEBPCYkwJFgUlhNKek6JPjSk7H0s05JITOkRMKhHMyWTom5NhqmApFimBKRYeUwQdiqkMCcbU1xDTUBeJdM0uhrSLxBNoakrpYyonjP2ScgZSTjmnNGTEjJHIeJTNJDM7UhSzHvTMQ1YpoQbh5NqcyFZ1Fg4VPsWZDZdS3E7A8TsppsSDk%2BKGfc7plzEXXMOSikZ3TMkvJyW8/Jny/kqMBWTdUKDkwkFZHCMFIKIXVLYJs%2BpsLGl7IxViwK8QTk4vOY83pmLkVcp5WkvlEz8WvWmUSiUEoi4xw4HHUgCck4pw4EEIIAA1ABf4ABK8Is7rDvNyfOhcFhLFAiASQGhRRmEkIkYJABOWh8QnVcH8voTgkh4691VbwBQegC6JyLnAWApcFTJEYmQCgEA/gRvoHEEeDhiCaL4HQVkOJKBREbrwfkzATTZtIH8CMAB5Bgtpe5YCdEYcQQbSD4EDF0O4/ruC8BPJ0FcGwm4UXlUnWGURUQmg8FgXuZ5njZqWARJgwAFAarwJgMExbkiMALfwFuohxAd1XV3FQ6ha26CSIPUAI8x76DwFEf1kAlioBVJkZtfqj4fmnu4c%2BEx6GBA5O/OYcRWH5HXpkTePDpDby1J%2B5eP6r4nyfi%2Btob6IPdCmKBoY4GoM5Bg9IV%2BTREPfqWEg7OeQzxI3HR6hV3ra2qsTXrN8DBQJOQgOqrVurYwVyIKCsw3JYweBeJGo1iwTVBrNaQb4j4hgQHNf4B1op/J2skKwjQDqeEOrMBoeIzlnLEa9Uqn1nA/UgASYGrQSwQ1QBgIgFA4bI3VxjeZ%2BNw9aDIC4PEDQCTa7pv9RALNLbSC5saPcAtRbGAEFLeW2tlbDAeI2Enetx9biYGbU3Nt%2BjWQFu7b3PtA77hDoi/nORLAiOTunbO%2Bdi7l2ea3a3Dd0gt2KB3b3fu%2Bgwu2ZPX2i9onk43oEHeie0XnAQFcJvSQ7qF53y/QkX9O9%2BvuuA5kLDo24N1BQxfAbXXr4IcXvfeIj9egTc25htbI34g4YNewAIBHO1ytjqRlVnAKMOdFNa6xdHNXat/Dqpj%2BAWNGtMpxuNbJc6OT4wZpYQmsBxFa/KjTuX/IJOVbwX1thdOkH08GkzJcQArAIMkFclnY2RryLCSucQAhlfXe3Srshqs91radgtYkmDJCI/KxVMPk6cGLSuTHo5MQ3fiHdu7etvvcb%2BwsAHUcxPxEkDa5ybCnNmB4bQ/y/lKjg94JD6HWmOA6b06asX1rZfi40J/Lg8muCsJYdIeV8RLuw%2B04j7XxGzBW5Zxr23/Glh4XSM4SQQA%3D)

```ll
; ModuleID = 'test_case'
source_filename = "test_case"
target datalayout = "e-i64:64-i128:128-v16:16-v32:32-n16:32:64"
target triple = "nvptx64-unknown-cuda-eabi"

@a0 = internal unnamed_addr global { i6, { [17 x i8] } } { i6 0, { [17 x i8] } undef }, align 4
@a_i8 = internal unnamed_addr global { i8, { [17 x i8] } } { i8 1, { [17 x i8] } undef }, align 4
@a_12 = internal unnamed_addr global { i6, { [16 x i8] } } { i6 1, { [16 x i8] } undef }, align 4

; free(): invalid pointer
;@a_invalid_pointer = internal unnamed_addr global { i6, { [17 x i8] } } { i6 1, { [17 x i8] } undef }, align 4

; malloc(): invalid size (unsorted)
@a_invalid_size_unsorted = internal unnamed_addr global { i6, { [36 x i8] } } { i6 1, { [36 x i8] } undef }, align 4
```

The actual struct that can't be compiled by LLVM 13 with PTX backend  is the `@builtin.os` global variable:

```ll
@builtin.os = internal unnamed_addr global { i6, [3 x i8], { [36 x i8] } } { i6 21, [3 x i8] undef, { [36 x i8] } undef }, align 4
```

I'll try to see how hard it would be to intercept it in the `link/NvPtx.zig`, if I can't manage that I'll just wait for LLVM 14 to drop which should happen in a few weeks IIUC.

**Comment by andrewrk** [comment]
Fix landed in feab1ebe1bfea3955320a6acba69f0c6c79c0730
